packages/massa-web3/src/experimental/smartContract.ts

```diff
@@ -192,6 +207,23 @@ export class SmartContract {
     return new Operation(this.client, await operation.send(details))
   }
 
+  async read(
+    func: string,
+    opts: ReadOnlyCallOptions
+  ): Promise<ReadOnlyCallResult> {
+    opts.maxGas = opts.maxGas ?? MAX_GAS_CALL
+
+    return await this.client.executeReadOnlyCall({
+      func,
+      parameter: opts.parameter,
+      targetAddress: this.contractAddress,
+      callerAddress: opts.callerAddress || this.contractAddress,
```

Comment:
I think that the default caller  address should be the wallet owner instead.

Reply:
Ok in that case we must pass an account or make the callerAddress not optional.

Reply:
But it is for the read function, right?

Reply:
You're right, this is not a clean alternative. How is it supposed to be used?